### PR TITLE
DM-40774: Create obs_rubinGenericCamera

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: doc/lsst.obs.rubinGenericCamera/index.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+   REPO=~/Repo/RubinGenericCamera		# N.b. not ~rlupton -- you own this file
+   LABEL="rhl-scr"
+   COLLBASE=u/rhl/tmp
+
+Clean up from previous attempts and initialise the ``butler``
+(in reality, I chain these blocks together with ``&&``)
+
+.. code-block:: sh
+
+   rm -rf $REPO &&
```
Comment (timj) [inline doc/lsst.obs.rubinGenericCamera/index.rst]: Is this missing a `butler create $REPO` somewhere?

File: doc/lsst.obs.rubinGenericCamera/testing.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+.. code-block:: bash
+
+   astrometadata dump myTestfile.fits > tests/headers/myTestfile.yaml
+
+
+CI Tests
+--------
+
+Integration tests are run from within the `ci_lsst <https://github.com/lsst-dm/ci_lsst>`_ package.
```
Comment (timj) [inline doc/lsst.obs.rubinGenericCamera/testing.rst]: I imagine this section needs to be removed or edited?

File: python/lsst/obs/rubinGenericCamera/rawFormatter.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+class RubinGenericCameraRawFormatter(FitsRawFormatterBase):
+    cameraClass = None
+    translatorClass = None
+    filterDefinitions = RUBIN_GENERIC_CAMERA_FILTER_DEFINITIONS
+
+    def getDetector(self, id):
+        return self.cameraClass().getCamera()[id]
+
+    def makeWcs(self, visitInfo, detector):
```
Comment (timj) [inline python/lsst/obs/rubinGenericCamera/rawFormatter.py]: I'm trying to work out how this differs from the base class implementation.

File: python/lsst/obs/rubinGenericCamera/translator.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+    default_resource_root = os.path.join(getPackageDir("obs_rubinGenericCamera"), "corrections")
+    """Default resource path root to use to locate header correction files."""
+
+    DETECTOR_MAX = 1
+
+    _const_map = {"detector_num": 0,
+                  "boresight_rotation_coord": "sky",
+                  "physical_filter": "empty",
+                  "detector_group": "None",
```
Comment (timj) [inline python/lsst/obs/rubinGenericCamera/translator.py]: This is a tricky one in that I'm clearly requiring it to be a string but there is no group of one detector. For LATISS we declare that the one detector is in group R00 to match the main camera.

File: python/lsst/obs/rubinGenericCamera/translator.py [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                  "relative_humidity": None,
```
Comment (timj) [inline python/lsst/obs/rubinGenericCamera/translator.py]: Your headers won't have weather information?

File: python/lsst/obs/rubinGenericCamera/translator.py [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+                  "pressure": None,
+                  "temperature": None,
+                  "focus_z": None,
+                  }
+    """Constant mappings"""
+
+    _trivial_map = {
+        "boresight_rotation_angle": (["ROTPA", "ROTANGLE"], dict(default=np.NaN, unit=u.deg)),
+        "detector_name": "CAMMM",
```
Comment (timj) [inline python/lsst/obs/rubinGenericCamera/translator.py]: I think this is meant to match the name of the detector that will be used in the detector dimension record (so CCD0) and not "AlliedVision GT3300" (but you could change the CCD0 name).

File: python/lsst/obs/rubinGenericCamera/translator.py [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+        return isStarTracker and camId == 102
+
+    @cache_translation
+    def to_instrument(self):
+        return "StarTrackerNarrw"
+
+    @cache_translation
+    def to_detector_serial(self):
+        return "00:0f:31:03:ae:60"
```
Comment (timj) [inline python/lsst/obs/rubinGenericCamera/translator.py]: I think it's meant to be allowed for const_map to work in subclasses so you can use:

```python
_const_map = {
    "detectors_serial": "00:0f:31:03:ae:60",
    "instrument": "StarTrackerNarrw",
}
```

File: ups/obs_rubinGenericCamera.table [hunk 8]
```diff
@@ -0,0 +1,8 @@
+setupRequired(afw)
```
Comment (timj) [inline ups/obs_rubinGenericCamera.table]: I don't think afw is used directly.

File: ups/obs_rubinGenericCamera.cfg [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# -*- python -*-
+
+import lsst.sconsUtils
```
Comment (timj) [inline ups/obs_rubinGenericCamera.cfg]: I think you can drop this file if you change the `SConstruct` file to use `noCfgFile=True`.

**Review by timj — APPROVED** [review]
This looks really good. Thanks.

File: doc/lsst.obs.rubinGenericCamera/index.rst (hunk 1 above)
Comment (RobertLuptonTheGood) [reply doc/lsst.obs.rubinGenericCamera/index.rst]: Lost in translation/cleanup.  Thanks

File: doc/lsst.obs.rubinGenericCamera/testing.rst (hunk 2 above)
Comment (RobertLuptonTheGood) [reply doc/lsst.obs.rubinGenericCamera/testing.rst]: I assumed that it was some automatic magic, but it sounds as if I should remove the section.

File: python/lsst/obs/rubinGenericCamera/rawFormatter.py (hunk 3 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/rubinGenericCamera/rawFormatter.py]: Good question.  Experimentally, we need `getDetector` but not `makeWcs`

File: python/lsst/obs/rubinGenericCamera/translator.py (hunk 4 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/rubinGenericCamera/translator.py]: If `"None"` is OK, I'll stick with it.

File: python/lsst/obs/rubinGenericCamera/translator.py (hunk 5 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/rubinGenericCamera/translator.py]: They don't currently.  Good point, I'll ask what our plans are, but for now this is correct.

**Pull request merged**